partcad/src/partcad/lint/lint.py

```
@@ -0,0 +1,107 @@
+        return report
+
+
+class Linting(ABC):
+    semaphore = None
+    MAX_CONCURRENT_CHECKS = None
+
+    def __init__(self, name: str) -> None:
+        self.name = name
+
+    @abstractmethod
+    async def validate(self, ctx: Context, package: Project, target: str, lint_ctx: dict = {}) -> LintingReport:
```

Ruff (0.11.9)

Do not use mutable data structures for argument defaults

Replace with `None`; initialize within function

(B006)